cirq-core/cirq/ops/phased_x_z_gate_test.py

```diff
@@ -53,9 +53,10 @@ def test_from_zyz_exponents(z0: float, y: float, z1: float) -> None:
     q = cirq.q(0)
     phxz = cirq.PhasedXZGate.from_zyz_exponents(z0, y, z1)
     zyz = cirq.Circuit(cirq.Z(q) ** z0, cirq.Y(q) ** y, cirq.Z(q) ** z1)
```

Comment:
lets keep the test as is and instead cast the exponents as floats (e.g. `cirq.Z(q) ** float(z0)`) and same goes for the other tests ... for example if we have a `gate ** 0` make it `gate ** 0.0` this should remove the need to update hte qasm test as well as other the tests

Reply:
If we do that, however, we end up in the same situation as before:
```python
import cirq
q = cirq.q(0)
op = cirq.X(q)

op * op != op ** 2

print(cirq.unitary(op * op))     # array([[1.+0.j]])
print(cirq.unitary(op ** 2))     # array([[1.+0.j, 0.+0.j], [0.+0.j, 1.+0.j]])
```

Reply:
just the tests

Reply:
For the moment I've changed the tests only, but I would say that the difference between exponentiation to int and float should be documented (if it remains like it is now).

If you want me to, I can modify the docstring of the pow function for paulis, describing the difference

Reply:
please do